draft-ietf-tls-esni.md

```diff
@@ -427,13 +439,13 @@ extensions MUST be ordered consecutively in ClientHelloInner. The list of outer
 extensions, OuterExtensions, includes those which were removed from
 EncodedClientHelloInner, in the order in which they were removed.
 
-Finally, EncodedClientHelloInner is serialized as a ClientHello structure,
-defined in Section 4.1.2 of {{RFC8446}}. Note this does not include the
-four-byte header included in the Handshake structure.
+Finally, the client sets the `padding` field to a byte string whose contents
+are all zeros. {{padding}} describes a recommended padding scheme.
 
 The client-facing server computes ClientHelloInner by reversing this process.
-First it makes a copy of EncodedClientHelloInner and copies the
-legacy_session_id field from ClientHelloOuter. It then looks for an
+First it parses EncodedClientHelloInner, interpreting all bytes after
+`client_hello` as padding. Next it makes a copy of the `client_hello` field and
```

Comment:
Should we require the server check that padding is all zeroes?

Reply:
Sure. Done.